SignalMessaging/environment/migrations/OWS109OutgoingMessageState.m

```diff
@@ -0,0 +1,81 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWS109OutgoingMessageState.h"
+#import <SignalServiceKit/OWSPrimaryStorage.h>
+#import <SignalServiceKit/TSOutgoingMessage.h>
+#import <YapDatabase/YapDatabaseTransaction.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+// Increment a similar constant for every future DBMigration
+static NSString *const OWS109OutgoingMessageStateMigrationId = @"109";
+
+@implementation OWS109OutgoingMessageState
+
++ (NSString *)migrationId
+{
+    return OWS109OutgoingMessageStateMigrationId;
+}
+
+// Override parent migration
+- (void)runUpWithCompletion:(OWSDatabaseMigrationCompletion)completion
+{
+    OWSAssert(completion);
+
+    OWSDatabaseConnection *dbConnection = (OWSDatabaseConnection *)self.primaryStorage.newDatabaseConnection;
+
+    [dbConnection asyncReadWriteWithBlock:^(YapDatabaseReadWriteTransaction *_Nonnull transaction) {
+        NSMutableArray<NSString *> *messageIds =
+            [[transaction allKeysInCollection:TSOutgoingMessage.collection] mutableCopy];
+        DDLogInfo(@"%@ Migrating %zd outgoing messages.", self.logTag, messageIds.count);
+
+        [self processBatch:messageIds
+              dbConnection:dbConnection
+                completion:^{
+                    DDLogInfo(@"Completed migration %@", self.uniqueId);
+
+                    [self save];
+
+                    completion();
+                }];
+    }];
```

Comment:
hmmm... I think we want to end this transaction *before* the process batch block get's called.

As written it might not make a difference since we're doing an asyncReadWrite, but seems unnecessarily dangerous.

Reply:
Done.